src/server.c

```diff
@@ -4709,7 +4706,7 @@ void getKeysSubcommandImpl(client *c, int with_flags) {
     if (!cmd) {
         addReplyError(c,"Invalid command specified");
         return;
-    } else if (cmd->getkeys_proc == NULL && cmd->key_specs_num == 0) {
+    } else if (!doesCommandHaveKeys(cmd)) {
```

Comment:
this is actually a bug fix?
previously it used to return `Invalid arguments specified for command` (since `getKeysFromCommandWithSpecs` would return 0), but now it'll return `The command has no key arguments`